### PR TITLE
feat: writing varnish chapter

### PR DESCRIPTION
**Pull request by alemorvan**
#### Author checklist (Completed by original Author)
- [x] Good fit for the Rocky Linux project? Title and Author Metatags inserted ?
- [x] If applicable, steps and instructions have been tested to work
- [x] Initial self-review to fix basic typos and grammar completed

#### Rocky Documentation checklist (Completed by Rocky team) 
- [x] 1st Pass (Document is good fit for project and author checklist completed)
- [x] 2nd Pass (Technical Review - check for technical correctness) 
- [x] 3rd Pass (Detailed Editorial Review and Peer Review)
- [x] Final approval (Final Review)

### CODE REVIEW

File: docs/books/web_services/05-load-balancer-proxies.md [hunk 1]
```diff
@@ -54,52 +55,695 @@ In this chapter, you will learn about XXXXXXX.
...
+sub vcl_backend_response {
+
+}
+
+sub vcl_deliver {
+
+}
+```
+
+* **vcl_recv**: This routine is called before the request is sent to the backend. In this routine, you can modify HTTP headers, cookies, choose the backend, etc. See actions `set req`..
```
Comment (SergeCroise) [inline docs/books/web_services/05-load-balancer-proxies.md:273]:
```suggestion
* **vcl_recv**: This routine is called before the request is sent to the backend. In this routine, you can modify HTTP headers, cookies, choose the backend, etc. See actions `set req`.
```

File: docs/books/web_services/05-load-balancer-proxies.md [hunk 2]
```diff
@@ -54,52 +55,695 @@ In this chapter, you will learn about XXXXXXX.
...
+
+By default, Varnish listens on port **6081** and uses **VCL** (**V**arnish **C**onfiguration **L**anguage) for its configuration. Thanks to VCL, it's possible to decide what should or shouldn't be transmitted to the client, what should be cached, from which site and how the response can be modified.
+
+Varnish can be extended with VMOD modules (Varnish Modules).
+
+#### Ensuring high availability
+
+Several mechanisms are used to ensure high availability throughout a web chain:
+
+* if varnish is behind loadbalancers: as the LBs are generaly in cluster mode, they are already in HA mode. A check from the LBs verifies varnish availability. If a varnish server no longer responds, it is automatically removed from the pool of available servers. In this case, varnish is in ACTIVE/ACTIVE mode.
```
Comment (SergeCroise) [inline docs/books/web_services/05-load-balancer-proxies.md:99]:
```suggestion
* if varnish is behind load balancers: as the LBs are generally in cluster mode, they are already in HA mode. A check from the LBs verifies varnish availability. If a varnish server no longer responds, it is automatically removed from the pool of available servers. In this case, varnish is in ACTIVE/ACTIVE mode.
```

File: docs/books/web_services/05-load-balancer-proxies.md [hunk 3]
```diff
@@ -54,52 +55,695 @@ In this chapter, you will learn about XXXXXXX.
...
+* if varnish isn't behind a LB cluster, clients address a VIP (see Heartbeat chapter), which is shared between the 2 varnishes. In this case, varnish is in ACTIVE/PASSIF mode. If the active server is no longer available, the VIP switches to the second varnish node.
```
Comment (SergeCroise) [inline docs/books/web_services/05-load-balancer-proxies.md:100]:
```suggestion
* if varnish isn't behind a LB cluster, clients address a VIP (see Heartbeat chapter), which is shared between the 2 varnishes. In this case, varnish is in ACTIVE/PASSIVE mode. If the active server is no longer available, the VIP switches to the second varnish node.
```

File: docs/books/web_services/05-load-balancer-proxies.md [hunk 4]
```diff
@@ -54,52 +55,695 @@ In this chapter, you will learn about XXXXXXX.
...
+
+To take advantage of the cache, the client must communicate with the web service on the default Varnish port 6081.
+
+![How Varnish works by default](img/varnish_website_with_varnish.png)
+
+To make the service transparent to the client, you'll need to change the default listening port for Varnish and the web service vhosts.
+
+![Transparent implementation for the customer](img/varnish_website_with_varnish_port_80.png)
+
+To provide an HTTPS service, you'll need to add either a loadbalancer upstream of the varnish service or a proxy service on the varnish server, such as Apache, Nginx or HAProxy.
```
Comment (SergeCroise) [inline docs/books/web_services/05-load-balancer-proxies.md:148]:
```suggestion
To provide an HTTPS service, you'll need to add either a load balancer upstream of the varnish service or a proxy service on the varnish server, such as Apache, Nginx or HAProxy.
```

File: docs/books/web_services/05-load-balancer-proxies.md [hunk 5]
```diff
@@ -54,52 +55,695 @@ In this chapter, you will learn about XXXXXXX.
...
+sub vcl_recv {
+    set req.backend = docs_director;
+}
+```
+
+#### Managing backends via CLI
+
+Backends can be marked as **sick** or **healthy** for administration or maintenance purposes. This action allows you to remove a node from the pool without having to modify the Varnish server configuration (and therefore without restarting it) or stop the backend service.
+
+View backend status : The `backend.list` command displays all backends, even those without a healthcheck (probe).
```
Comment (SergeCroise) [inline docs/books/web_services/05-load-balancer-proxies.md:446]:
```suggestion
View backend status : The `backend.list` command displays all backends, even those without a health check (probe).
```

**Comment by alemorvan** [comment]
Thx @SergeCroise

**Pull request merged**